src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -277,6 +277,49 @@ public TimeSpan Expect100ContinueTimeout
             }
         }
 
+        public TimeSpan KeepAlivePingDelay
+        {
+            get => _settings._keepAlivePingDelay;
+            set
+            {
+                if ((value < TimeSpan.Zero && value != Timeout.InfiniteTimeSpan) ||
```

Comment:
Timeout.InfiniteTimeSpan should equal off

Zero shouldn't be allowed. A keep alive ping of zero doesn't make sense.

In ASP.NET Core the minimum allow value is 1 second. Partly we did that because there is a global server heartbeat of once a second and that is the lowest resolution allowed, but also because pinging more often than once a second doesn't make sense. Why would someone want to set `KeepAlivePingDelay = TimeSpan.FromMilliseconds(1)`?

Reply:
I don't know if it makes to automatically clip the value for the user, but if we did then something like 10 seconds seems very reasonably more frequent than anyone would want to use.

Reply:
0 and Timeout.InfiniteTimeSpan means we don't ping at all. It is default behavior for SocketsHttpHandler and the ping should be enabled explicitly by this property

Reply:
The default was changed to `Timeout.InfiniteTimeSpan` and 0 is forbidden now